### PR TITLE
Guard against a slow scheduler with backpressure strategy.

### PR DESCRIPTION
**Pull request by JakeWharton**
Schedulers apply backpressure upstream which previously behaved as an unbounded stream. When the scheduler worker couldn't keep up an MBPE was thrown. We already use the 'latest' strategy for slowing 'Query' emissions after the scheduler so we can apply the same thing before it.

This comes with a subtle change in that transactions will now synchronously run every filter Func before continuing. Previously the filters ran on the scheduler and did not block the thread performing the transaction. This means that the more queries that are listening at once, the slower every transaction's completing step will be. I don't expect this to have any meaningful impact since it's very fast and happening on a background thread already.

### CODE REVIEW

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BriteContentResolver.java [hunk 1]
```diff
@@ -120,9 +120,10 @@ public QueryObservable createQuery(@NonNull final Uri uri, @Nullable final Strin
       }
     };
     Observable<Query> queryObservable = Observable.create(subscribe) //
+        .onBackpressureLatest() // Guard against uncontrollable frequency of upstream emissions.
```
Comment (AlecStrong) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteContentResolver.java]: just curious...why is this here instead of before the `observeOn`? I thought that since it was the scheduler that couldn't keep up with the emissions it would need the backpressure applied immediately before.

**Comment by AlecStrong** [comment]
LGTM

File: sqlbrite/src/main/java/com/squareup/sqlbrite/BriteContentResolver.java (hunk 1 above)
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/BriteContentResolver.java]: It doesn't matter. `startWith` only emits one item and then defers items to the upstream observable. The only case where you'd actually get a backpressure problem with a single item is if the queue size was 0, which is impossible.

My next PR will eliminate `startWith` in this case anyway and move it into the upstream.
